pipelines/batch/src/main/java/org/openmrs/analytics/JdbcFetchUtil.java

```diff
@@ -253,4 +313,37 @@ public Map<String, List<String>> createFhirReverseMap(String resourceString, Dat
 		}
 		return reverseMap;
 	}
+	
+	/**
+	 * Generates the query parameters by evenly distributing the workload to each query. The number of
+	 * resources of each resource type is taken into consideration with respect to the total number of
+	 * resources to be fetched inorder to determine the porportion of JdbcIo workers allocated.
+	 */
+	@VisibleForTesting
+	List<List<String>> generateQueryParameters(Pipeline pipeline, String resourceList,
+	        HashMap<String, Integer> resourceCount, int jdbcMaxPoolSize) {
+		List<List<String>> queryParameterList = new ArrayList<List<String>>();
+		
+		int totalCount = 0;
+		for (int count : resourceCount.values()) {
+			totalCount += count;
+		}
+		
+		for (String resourceType : resourceList.split(",")) {
+			int numProcsAllocated = (int) Math.ceil((double) resourceCount.get(resourceType) / totalCount * jdbcMaxPoolSize);
+			System.out.println(resourceType);
```

Comment:
Please remove `System` prints and replace them with logs if needed (here and everywhere else).